python/adelphi/adelphi/nb.py

```diff
@@ -0,0 +1,127 @@
+import yaml
+
+from itertools import chain
+
+from adelphi.exceptions import TooManyKeyspacesException, TooManyTablesException
+from adelphi.export import BaseExporter
+
+SEQS={}
+SEQS["text"] = "Mod(1000000000); ToString() -> String"
+DISTS={}
+DISTS["text"] = "Hash(); Mod(1000000000); ToString() -> String"
+
+# A collection of functionally static functions
+def dist_binding_name(col):
+    return "{}_dist".format(col.name)
+
+
+def seq_binding_name(col):
+    return "{}_seq".format(col.name)
+
+
+def build_bindings(table):
+    rv = {}
+    for (col_name, col) in table.columns.items():
+        if col.cql_type not in DISTS.keys():
+            raise UnsupportedColumnTypeException("No nosqlbench distribution configured for type {}".format(col.cql_type))
+        rv[dist_binding_name(col)] = DISTS[col.cql_type]
+
+        # If we have a primary key we'll also need a seq for insert statements
+        if col_name in [c.name for c in table.primary_key]:
+            if col.cql_type not in SEQS.keys():
+                raise UnsupportedColumnTypeException("No nosqlbench sequence configured for type {}".format(col.cql_type))
+            rv[seq_binding_name(col)] = SEQS[col.cql_type]
+
+    return rv
+
+
+def build_select_statements(keyspace, table):
+    key_bindings = " and ".join(["{} = {}".format(key.name, "{" + dist_binding_name(key) + "}") for key in table.primary_key])
+    return "select * from  {}.{} where {}".format(keyspace.name, table.name, key_bindings)
+
+
+def build_insert_statements(keyspace, table):
+    cols = table.columns.values()
+    primary_keys = set(table.primary_key)
+    col_names = ",".join([col.name for col in cols])
+    def binding_name(col):
+        return seq_binding_name(c) if col in primary_keys else dist_binding_name(c)
+    col_bindings = ",".join(["{" + binding_name(c) + "}" for c in cols])
+    return "insert into {}.{} ({}) values ({})".format(keyspace.name, table.name, col_names, col_bindings)
+
+
+class UnsupportedColumnTypeException(Exception):
+    """Exception indicinating that we can't create nosqlbench structures for a column type"""
+    def __init__(self, msg):
+        self.message = msg
+
+
+class NbExporter(BaseExporter):
+
+    def __init__(self, cluster, props):
+
+        # Always disable anonymization when generating nosqlbench configs
+        self.props = props.copy()
+        self.props["anonymize"] = False
+
+        self.metadata = self.get_common_metadata(cluster, self.props)
+
+        all_keyspaces = self.get_keyspaces(cluster, self.props)
+        if len(all_keyspaces) > 1:
+            raise TooManyKeyspacesException
```

Comment:
Seemed like a reasonable compromise for now.  We could create distinct nosqlbench configs for each input keyspace (along the lines of what we do for Gemini) but that felt pretty artificial.  And we don't yet have a clear understanding of what a nosqlbench config for multiple keyspaces looks like... do we want to generate select/insert statements for everything in _all_ the keyspaces in such a case?

Seemed safer to go with the minimal path for now.